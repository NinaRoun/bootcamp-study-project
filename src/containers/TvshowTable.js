import React, { Component } from 'react';
import connect from "react-redux/es/connect/connect";
import { fetchTvshows } from '../actions'
import SortingCriteria from '../components/SortingCriteria';
import TvshowItem from '../components/TvshowItem';
import Pagination from '../components/Pagination';
import Search from '../components/Search';


class TvshowTable extends Component {

    constructor(props) {
        super(props)
        this.state = {
            currentPage: 1,
            showsPerPage: 6,
            search: "",
            pageNumbers: []
        }

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(page) {
        //console.log('current page is ', this.state.currentPage););
        this.setState({
            currentPage: Number(page)
        })
    }

    handleChange(event) {
        this.setState({
            search: event.target.value
        })

        const pageNumbers = [];
        const tableItems = document.getElementsByTagName('tr');
        console.log('length ' + tableItems.length);
        console.log('search = ' + this.state.search);
        for (let i = 1; i <= Math.ceil(tableItems.length / this.state.showsPerPage); i++) {
            pageNumbers.push(i);
        }
        this.setState({
            pageNumbers
        })

    }

    componentWillMount() {
        fetchTvshows()(this.props.dispatch);
    }

    countPages() {
        const pageNumbers = [];
        const tableItems = document.getElementsByTagName('tr');
            console.log('length ' + tableItems.length);
            console.log('search = ' + this.state.search);
            for (let i = 1; i <= Math.ceil(tableItems.length / this.state.showsPerPage); i++) {
                pageNumbers.push(i);
            }
        this.setState({
                    pageNumbers
                })
    }

    render() {

        const {currentPage, showsPerPage} = this.state;

        const indexOfLastShow = currentPage * showsPerPage;  //6, 12, 18, ...
        const indexOfFirstShow = indexOfLastShow - showsPerPage; //0, 6, 12, ...
        const currentShows = this.props.tvshows.slice(indexOfFirstShow, indexOfLastShow); //here: per 6 on each page

        const tvshowItems = this.props.tvshows && currentShows
            .filter(tvshow => !this.state.search ||
                tvshow.name.toLowerCase().indexOf(this.state.search.toLowerCase()) > -1 ||
                tvshow.overview.toLowerCase().indexOf(this.state.search.toLowerCase()) > -1
            )
            .map(
            (tvshow, index) => <TvshowItem key={tvshow.id} tvshow={tvshow}/>
        );

        const pageNumbers = [];
        //for !this.state.search
        console.log('no search')
        for (let i = 1; i <= Math.ceil(this.props.tvshows.length / showsPerPage); i++) { //here: 4 pages
            pageNumbers.push(i);
        }

        const renderPageNumbers = !this.state.search.length ? pageNumbers.map(number => {
            return (
                <Pagination
                    key={number}
                    id={number}
                    number={number}
                    onClick={this.handleClick}
                    className={number === this.state.currentPage ? "active" : ""}
                />)}) :
            this.state.pageNumbers.map(number => {
                return (
                    <Pagination
                        key={number}
                        id={number}
                        number={number}
                        onClick={this.handleClick}
                        className={number === this.state.currentPage ? "active" : ""}
                    />)})

        return (
            <div className="mainTable">

                <div className="intro">
                    <SortingCriteria />
                    < Search onChange={this.handleChange.bind(this)}/>
                </div>

                <table>
                    <tbody>

                    {tvshowItems}

                    </tbody>
                </table>

                <div id="page-numbers">
                    {renderPageNumbers}
                </div>

            </div>
        );
    }
}

const mapStateToProps = state => ({
    tvshows: state.tvshows
});

const mapDispatchToProps = dispatch => ({
    dispatch,
})

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(TvshowTable, Pagination)

//не могу сделать нормальную пагинацию, уже всяко разно проовала. Думала о такой логике: для search есть handleChange. Каждый раз, когда происходит изменение, происходит перерасчет страниц.
//хотела подсчитывать количество тегов tr и делать что-то вроде:
//const tableItems = document.getElementsByTagName('tr');
//     for (let i = 1; i <= Math.ceil(tableItems.length / showsPerPage); i++) {
//         pageNumbers.push(i);
//     }
//но тогда придется pageNumbers держать в state , выщитывая количество страниц в первый раз, тоже