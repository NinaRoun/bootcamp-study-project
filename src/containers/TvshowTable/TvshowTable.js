import React, { Component } from 'react';
import { connect } from "react-redux";
import { fetchTvshows } from '../../actions'
import SortingCriteria from '../../components/SortingCriteria/SortingCriteria';
import TvshowItem from '../../components/TvshowItem/TvshowItem';
import Pagination from '../../components/Pagination/Pagination';
import Search from '../../components/Search/Search';
import Loading from '../../components/Loading/Loading';
import Error from '../../components/Error/Error';
import style from '../../stylesheets/main.styl';

export class TvshowTable extends Component {

    constructor(props) {
        super(props);
        this.state = {
            currentPage: 1,
            showsPerPage: 6,
            search: "",
            pageNumbers: [],
        };

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(page) {
        this.setState({
            currentPage: Number(page)
        })
    }

    handleChange(event) {
        this.setState({
            search: event.currentTarget.value
        })
    }

    componentWillMount() {
        this.props.fetchTvShows();
    }

    render() {

        const {currentPage, showsPerPage, search} = this.state;

        const allShows = !this.state.search ? this.props.tvshows : this.props.tvshows.filter(tvshow => !this.state.search ||
            tvshow.name.toLowerCase().indexOf(this.state.search.toLowerCase()) > -1 ||
            tvshow.overview.toLowerCase().indexOf(this.state.search.toLowerCase()) > -1
        );

        const indexOfLastShow = currentPage * showsPerPage;  //6, 12, 18, ...
        const indexOfFirstShow = indexOfLastShow - showsPerPage; //0, 6, 12, ...
        const currentShows = allShows.slice(indexOfFirstShow, indexOfLastShow); //here: per 6 on each page

        const tvshowItems = this.props.tvshows && currentShows
            .map(
            (tvshow, index) => <TvshowItem key={tvshow.id} tvshow={tvshow} search={search}/>
        );

        const pageNumbers = [];
        for (let i = 1; i <= Math.ceil(allShows.length / showsPerPage); i++) { //here: 4 pages
            pageNumbers.push(i);
        }

        const renderPageNumbers = pageNumbers.map(number => {
            return (
                <Pagination
                    key={number}
                    id={number}
                    number={number}
                    onClick={this.handleClick}
                    className={number === this.state.currentPage ? style.active : ""}
                />)});

        return (
            <div className={style.mainTable}>

                {this.props.isLoading ? <Loading /> : this.props.error ? <Error error={this.props.error} /> :

                    <div className="pageContent">
                        <div className={style.options}>
                            <SortingCriteria />
                            <Search onChange={this.handleChange.bind(this)}/>
                        </div>

                        <table>
                            <tbody>

                            {tvshowItems}

                            </tbody>
                        </table>

                        <div id={style.pageNumbers}>
                            {renderPageNumbers}
                        </div>
                    </div>

                }

            </div>
        );
    }
}

const mapStateToProps = state => ({
    tvshows: state.tvshowsData.tvshows,
    isLoading: state.tvshowsData.isLoading,
    error: state.tvshowsData.error
});

const mapDispatchToProps = dispatch => ({
    fetchTvShows: () => dispatch(fetchTvshows())
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(TvshowTable, Pagination)

