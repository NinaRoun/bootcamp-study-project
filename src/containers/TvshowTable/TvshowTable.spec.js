import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { TvshowTable } from './TvshowTable.js';

configure({ adapter: new Adapter() });

function setup() {
    const props = {
        tvshows: [{
                id: 1,
                name: 'one',
                origin_country: 'US',
                original_language: 'English',
                overview: 'Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.',
                vote_average: 7.3,
                vote_count: 250,
                first_air_date: "2020-03-21"
            },{
                id: 2,
                name: 'two',
                origin_country: 'US',
                original_language: 'English',
                overview: 'Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.',
                vote_average: 7.3,
                vote_count: 250,
                first_air_date: "2020-03-21"
            }],
        isLoading: false,
        error: "",
        fetchTvShows: () => {}
    };

    return {
        props
    }
}

describe('Tvshow container initial', () => {
    const mockFetchGetShows = jest.fn();
    const { props } = setup();
    const nextProps = {
        ...props,
        fetchTvShows: mockFetchGetShows
    };
    const enzymeWrapper = shallow(<TvshowTable {...nextProps} />);

    it('renders properly', () => {
        expect(enzymeWrapper).toMatchSnapshot();
    });

    it('calls the `fetchTvShows` function', () => {
        expect(mockFetchGetShows).toHaveBeenCalledTimes(1)
    });
});

describe('Tvshow preloader', () => {
    const { props } = setup();
    const nextProps = {
        ...props,
        isLoading: true
    };
    const enzymeWrapper = shallow(<TvshowTable {...nextProps} />);

    it('renders properly', () => {
        const loadingComponent = enzymeWrapper.find('Loading');
        expect(loadingComponent).toHaveLength(1);
    });
});

describe('Error while fetching', () => {
    const { props } = setup();
    const nextProps = {
        ...props,
        error: '500 server error'
    };
    const enzymeWrapper = shallow(<TvshowTable {...nextProps} />);

    it('renders properly', () => {
        const errorComponent = enzymeWrapper.find('Error');
        expect(errorComponent).toHaveLength(1);
    });
});



