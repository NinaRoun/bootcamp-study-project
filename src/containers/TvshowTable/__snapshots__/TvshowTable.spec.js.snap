// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tvshow container initial renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TvshowTable
    error=""
    fetchTvShows={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    isLoading={false}
    tvshows={
      Array [
        Object {
          "first_air_date": "2020-03-21",
          "id": 1,
          "name": "one",
          "origin_country": "US",
          "original_language": "English",
          "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
          "vote_average": 7.3,
          "vote_count": 250,
        },
        Object {
          "first_air_date": "2020-03-21",
          "id": 2,
          "name": "two",
          "origin_country": "US",
          "original_language": "English",
          "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
          "vote_average": 7.3,
          "vote_count": 250,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="pageContent"
      >
        <div>
          <SortingCriteria />
          <Search
            onChange={[Function]}
          />
        </div>
        <table>
          <tbody>
            <TvshowItem
              search=""
              tvshow={
                Object {
                  "first_air_date": "2020-03-21",
                  "id": 1,
                  "name": "one",
                  "origin_country": "US",
                  "original_language": "English",
                  "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                  "vote_average": 7.3,
                  "vote_count": 250,
                }
              }
            />
            <TvshowItem
              search=""
              tvshow={
                Object {
                  "first_air_date": "2020-03-21",
                  "id": 2,
                  "name": "two",
                  "origin_country": "US",
                  "original_language": "English",
                  "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                  "vote_average": 7.3,
                  "vote_count": 250,
                }
              }
            />
          </tbody>
        </table>
        <div>
          <Pagination
            id={1}
            number={1}
            onClick={[Function]}
          />
        </div>
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <SortingCriteria />
            <Search
              onChange={[Function]}
            />
          </div>,
          <table>
            <tbody>
              <TvshowItem
                search=""
                tvshow={
                  Object {
                    "first_air_date": "2020-03-21",
                    "id": 1,
                    "name": "one",
                    "origin_country": "US",
                    "original_language": "English",
                    "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                    "vote_average": 7.3,
                    "vote_count": 250,
                  }
                }
              />
              <TvshowItem
                search=""
                tvshow={
                  Object {
                    "first_air_date": "2020-03-21",
                    "id": 2,
                    "name": "two",
                    "origin_country": "US",
                    "original_language": "English",
                    "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                    "vote_average": 7.3,
                    "vote_count": 250,
                  }
                }
              />
            </tbody>
          </table>,
          <div>
            <Pagination
              id={1}
              number={1}
              onClick={[Function]}
            />
          </div>,
        ],
        "className": "pageContent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <SortingCriteria />,
              <Search
                onChange={[Function]}
              />,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tbody>
              <TvshowItem
                search=""
                tvshow={
                  Object {
                    "first_air_date": "2020-03-21",
                    "id": 1,
                    "name": "one",
                    "origin_country": "US",
                    "original_language": "English",
                    "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                    "vote_average": 7.3,
                    "vote_count": 250,
                  }
                }
              />
              <TvshowItem
                search=""
                tvshow={
                  Object {
                    "first_air_date": "2020-03-21",
                    "id": 2,
                    "name": "two",
                    "origin_country": "US",
                    "original_language": "English",
                    "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                    "vote_average": 7.3,
                    "vote_count": 250,
                  }
                }
              />
            </tbody>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <TvshowItem
                  search=""
                  tvshow={
                    Object {
                      "first_air_date": "2020-03-21",
                      "id": 1,
                      "name": "one",
                      "origin_country": "US",
                      "original_language": "English",
                      "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                      "vote_average": 7.3,
                      "vote_count": 250,
                    }
                  }
                />,
                <TvshowItem
                  search=""
                  tvshow={
                    Object {
                      "first_air_date": "2020-03-21",
                      "id": 2,
                      "name": "two",
                      "origin_country": "US",
                      "original_language": "English",
                      "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                      "vote_average": 7.3,
                      "vote_count": 250,
                    }
                  }
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {
                  "search": "",
                  "tvshow": Object {
                    "first_air_date": "2020-03-21",
                    "id": 1,
                    "name": "one",
                    "origin_country": "US",
                    "original_language": "English",
                    "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                    "vote_average": 7.3,
                    "vote_count": 250,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "function",
                "props": Object {
                  "search": "",
                  "tvshow": Object {
                    "first_air_date": "2020-03-21",
                    "id": 2,
                    "name": "two",
                    "origin_country": "US",
                    "original_language": "English",
                    "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                    "vote_average": 7.3,
                    "vote_count": 250,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "tbody",
          },
          "type": "table",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Pagination
                id={1}
                number={1}
                onClick={[Function]}
              />,
            ],
            "id": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "className": undefined,
                "id": 1,
                "number": 1,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="pageContent"
        >
          <div>
            <SortingCriteria />
            <Search
              onChange={[Function]}
            />
          </div>
          <table>
            <tbody>
              <TvshowItem
                search=""
                tvshow={
                  Object {
                    "first_air_date": "2020-03-21",
                    "id": 1,
                    "name": "one",
                    "origin_country": "US",
                    "original_language": "English",
                    "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                    "vote_average": 7.3,
                    "vote_count": 250,
                  }
                }
              />
              <TvshowItem
                search=""
                tvshow={
                  Object {
                    "first_air_date": "2020-03-21",
                    "id": 2,
                    "name": "two",
                    "origin_country": "US",
                    "original_language": "English",
                    "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                    "vote_average": 7.3,
                    "vote_count": 250,
                  }
                }
              />
            </tbody>
          </table>
          <div>
            <Pagination
              id={1}
              number={1}
              onClick={[Function]}
            />
          </div>
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <SortingCriteria />
              <Search
                onChange={[Function]}
              />
            </div>,
            <table>
              <tbody>
                <TvshowItem
                  search=""
                  tvshow={
                    Object {
                      "first_air_date": "2020-03-21",
                      "id": 1,
                      "name": "one",
                      "origin_country": "US",
                      "original_language": "English",
                      "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                      "vote_average": 7.3,
                      "vote_count": 250,
                    }
                  }
                />
                <TvshowItem
                  search=""
                  tvshow={
                    Object {
                      "first_air_date": "2020-03-21",
                      "id": 2,
                      "name": "two",
                      "origin_country": "US",
                      "original_language": "English",
                      "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                      "vote_average": 7.3,
                      "vote_count": 250,
                    }
                  }
                />
              </tbody>
            </table>,
            <div>
              <Pagination
                id={1}
                number={1}
                onClick={[Function]}
              />
            </div>,
          ],
          "className": "pageContent",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <SortingCriteria />,
                <Search
                  onChange={[Function]}
                />,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tbody>
                <TvshowItem
                  search=""
                  tvshow={
                    Object {
                      "first_air_date": "2020-03-21",
                      "id": 1,
                      "name": "one",
                      "origin_country": "US",
                      "original_language": "English",
                      "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                      "vote_average": 7.3,
                      "vote_count": 250,
                    }
                  }
                />
                <TvshowItem
                  search=""
                  tvshow={
                    Object {
                      "first_air_date": "2020-03-21",
                      "id": 2,
                      "name": "two",
                      "origin_country": "US",
                      "original_language": "English",
                      "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                      "vote_average": 7.3,
                      "vote_count": 250,
                    }
                  }
                />
              </tbody>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <TvshowItem
                    search=""
                    tvshow={
                      Object {
                        "first_air_date": "2020-03-21",
                        "id": 1,
                        "name": "one",
                        "origin_country": "US",
                        "original_language": "English",
                        "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                        "vote_average": 7.3,
                        "vote_count": 250,
                      }
                    }
                  />,
                  <TvshowItem
                    search=""
                    tvshow={
                      Object {
                        "first_air_date": "2020-03-21",
                        "id": 2,
                        "name": "two",
                        "origin_country": "US",
                        "original_language": "English",
                        "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                        "vote_average": 7.3,
                        "vote_count": 250,
                      }
                    }
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "function",
                  "props": Object {
                    "search": "",
                    "tvshow": Object {
                      "first_air_date": "2020-03-21",
                      "id": 1,
                      "name": "one",
                      "origin_country": "US",
                      "original_language": "English",
                      "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                      "vote_average": 7.3,
                      "vote_count": 250,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "function",
                  "props": Object {
                    "search": "",
                    "tvshow": Object {
                      "first_air_date": "2020-03-21",
                      "id": 2,
                      "name": "two",
                      "origin_country": "US",
                      "original_language": "English",
                      "overview": "Lorem ipsum dolor sit amet, searchedArea consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.",
                      "vote_average": 7.3,
                      "vote_count": 250,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "tbody",
            },
            "type": "table",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Pagination
                  id={1}
                  number={1}
                  onClick={[Function]}
                />,
              ],
              "id": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {
                  "className": undefined,
                  "id": 1,
                  "number": 1,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
